misc(syncer): make timeout configurable

AryanGodara opened the pull request:
<!--
Please read and fill out this form before submitting your PR.

Please make sure you have reviewed our contributors guide before submitting your
first PR.
-->

## Overview
Resolves #155
<!-- 
Please provide an explanation of the PR, including the appropriate context,
background, goal, and rationale. If there is an issue with this information,
please provide a tl;dr and link the issue. 
-->

## Checklist

<!-- 
Please complete the checklist to ensure that the PR is ready to be reviewed.

IMPORTANT:
PRs should be left in Draft until the below checklist is completed.
-->

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

AryanGodara commented:
@vgonkivs This one's ready for review :)

Wondertan reviewed (changes requested):
This is wrong. The state struct is not supplied by under to the Syncer, but the Syncer provides it to user through State method on Syncer. Therefore, you can't configure anything though it.

Instead, we should add this configuration field to Config

vgonkivs commented:
hey @AryanGodara, thanks for you contribution. Let me help you to improve your code: If you take a look [here](https://github.com/celestiaorg/go-header/blob/e2197d9350821b3bc970eec236d5cbe9c45e711e/sync/options.go#L13), you will see the Parameter struct that allows configuring the syncer. In the syncer c-tor we create default parameters and override them using `options` pattern, which you can find in the same files I've added previously. The idea is to add `maxAwait` to the Parameter struct + set the default value in `DefaultParameters` and add the ability to override the default maxAwait time.

AryanGodara commented:
> hey @AryanGodara, thanks for you contribution. Let me help you to improve your code: If you take a look [here](https://github.com/celestiaorg/go-header/blob/e2197d9350821b3bc970eec236d5cbe9c45e711e/sync/options.go#L13), you will see the Parameter struct that allows configuring the syncer. In the syncer c-tor we create default parameters and override them using `options` pattern, which you can find in the same files I've added previously. The idea is to add `maxAwait` to the Parameter struct + set the default value in `DefaultParameters` and add the ability to override the default maxAwait time.

Wow thank you for explaining it in such detail @vgonkivs :D
This was extremely helpful. I've just pushed new changes. I hope this time, I've implemented it correctly (:

renaynay commented:
Hi @AryanGodara thank you so much for your contribution. We've decided that we don't actually need this feature unfortunately. I apologise for the last minute change of mind. There are many other good first issues that we welcome you to work on, however!

AryanGodara commented:
> Hi @AryanGodara thank you so much for your contribution. We've decided that we don't actually need this feature unfortunately. I apologise for the last minute change of mind. There are many other good first issues that we welcome you to work on, however!

It's quite alright @renaynay :D
It was a good issue to get onboarded to the project. I hope I'll continue making new good contributions over time :D
